code/modules/clothing/glasses/glasses.dm

```diff
@@ -257,6 +257,9 @@
 		if(noir_mode)
 			if(color_view && user.client && !user.client.color)
 				animate(user.client, color = color_view, time = 10)
+	else
+		if(user.client && user.client.color)

```

Comment:
this should probably be `if(user.client && user.client.color == color_view)`, in case we ever make more color views for clients